privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -626,11 +626,11 @@ def getUserId(self, LoginName):
                             LoginName))
 
         for entry in r:
-            userid = self._get_uid(entry, self.uidtype)
+            userid = str(self._get_uid(entry, self.uidtype))
```

Comment:
I guess we need to use ``six.text_type`` here (or ``to_unicode``?). If we use ``str()``, this breaks under Python 2.7 with uids that contain non-ASCII characters (like ``dn`` with an umlaut):
```
Traceback (most recent call last):
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask/app.py", line 1947, in full_dispatch_request
    rv = self.preprocess_request()
  File "/home/fred/privacyidea/privacyidea/venv/lib/python2.7/site-packages/flask/app.py", line 2241, in preprocess_request
    rv = func()
  File "/home/fred/privacyidea/privacyidea/privacyidea/api/auth.py", line 350, in decorated_function
    return f(*args, **kwargs)
  File "/home/fred/privacyidea/privacyidea/privacyidea/api/before_after.py", line 99, in before_user_request
    before_request()
  File "/home/fred/privacyidea/privacyidea/privacyidea/api/before_after.py", line 143, in before_request
    request.User = get_user_from_param(request.all_data)
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/log.py", line 154, in log_wrapper
    return func(*args, **kwds)
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/user.py", line 582, in get_user_from_param
    resolver=param.get("resolver"))
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/log.py", line 154, in log_wrapper
    return func(*args, **kwds)
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/user.py", line 98, in __init__
    self._get_user_from_userstore()
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/usercache.py", line 63, in cache_wrapper
    return wrapped_function(*args, **kwds)
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/user.py", line 106, in _get_user_from_userstore
    self._get_resolvers()
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/user.py", line 207, in _get_resolvers
    if self._locate_user_in_resolver(resolvername):
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/user.py", line 227, in _locate_user_in_resolver
    uid = y.getUserId(self.login)
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/resolvers/LDAPIdResolver.py", line 258, in cache_wrapper
    f_result = func(self, *args, **kwds)
  File "/home/fred/privacyidea/privacyidea/privacyidea/lib/resolvers/LDAPIdResolver.py", line 629, in getUserId
    userid = str(self._get_uid(entry, self.uidtype))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xf6' in position 14: ordinal not in range(128)
```

Reply:
Since we do the unicode conversion in LDAP-Resolver now, this should be fixed in 61743ea